### PR TITLE
Add JSON schema validation for Prompt Editor

### PR DESCRIPTION
**Pull request by CoenWarmer**
## Summary

This adds JSON validation to the Prompt Editor according to a supplied JSON function schema.

It also cleans up unused hooks such as `useFunctions` as that is being supplied by the ObsAiAssistant Service.

### CODE REVIEW

File: x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export const useJsonEditorModel = (functionDefinition?: FunctionDefinition) => {
+  return useMemo(() => {
+    if (!functionDefinition) {
+      return {};
+    }
+
+    const schema = { ...functionDefinition.options.parameters };
+
+    const initialJsonString = functionDefinition.options.parameters.properties
+      ? Object.keys(functionDefinition.options.parameters.properties).reduce(
```
Comment (CoenWarmer) [inline x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts:25]: This currently assumes properties is a flat object. Not quite sure if that is the case always. Cc @dgieselaar

Comment (dgieselaar) [reply x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts:25]: It's not. You should be able to tell from the schema whether to create an object or an empty string. I think it's a nice addition but if it's too complicated to get a correct initial value we should just stick to an empty object (so `{}`).

File: x-pack/plugins/observability_ai_assistant/public/components/chat/function_list_popover.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  };
+
+  const handleSelectFunction = (func: FunctionDefinition) => {
+    setIsFunctionListOpen(false);
+    onSelectFunction(func);
+  };
+
+  useEffect(() => {
+    const keyboardListener = (event: KeyboardEvent) => {
+      if (event.code === 'Digit4') {
```
Comment (dgieselaar) [inline x-pack/plugins/observability_ai_assistant/public/components/chat/function_list_popover.tsx:42]: just the 4?

**Review by dgieselaar — APPROVED** [review]

File: x-pack/plugins/observability_ai_assistant/public/components/chat/function_list_popover.tsx (hunk 2 above)
Comment (CoenWarmer) [reply x-pack/plugins/observability_ai_assistant/public/components/chat/function_list_popover.tsx:42]: Good call

File: x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts (hunk 1 above)
Comment (CoenWarmer) [reply x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts:25]: I think it's cool addition that makes the experience a bit nicer. Will pick this up in a new PR.

**Pull request merged**